functions/PayrollProcessor.Functions.Seeding/Infrastructure/DomainFaker.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Bogus;
+
+namespace PayrollProcessor.Functions.Seeding.Infrastructure
+{
+    /// <summary>
+    /// Creates Fakers that use a specific constructor
+    /// See: https://github.com/bchavez/Bogus/issues/291#issuecomment-614371450
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class DomainFaker<T> : Faker<T> where T : class
+    {
+        public DomainFaker(ConstructorInfo constructor, Func<T> factory) =>
```

Comment:
This class helps create instances of types that have required constructor parameters.

We can definitely make this more intelligent in the future if the domain types become more strict.

We could also switch from using domain types to request types or table entity types that would inherently have parameterless constructors.